app/View/Users/logout401.ctp

```diff
@@ -0,0 +1,41 @@
+<div style="width:100%;">
+    <script>
+        // Chrome/Edge will log you out once it sees the HTTP401.
+        // We need to be extra hacky to properly log out on i.e. Firefox.
+        <?php
+        $split_baseurl = array();
+        # We split the baseurl, since we need to add data between the
+        # schema and hostname. We *could* use parse_url here, but then
+        # we would need a lot of code to rebuild it
+        if (preg_match("/(https?:\/\/)(.*)/", $baseurl, $split_baseurl)):
+        ?>
+            let logoutxhr401 = new XMLHttpRequest()
+            logoutxhr401.open("GET", "<?php echo $split_baseurl[1]; ?>logout:@<?php echo $split_baseurl[2]; ?>/users/login")
```

Comment:
I'm not sure I understand what's going on here. Do you mind to explain why you need to perform a self call that would look like this `https://logout:@misp-hostname.org/users/login` ?
Is LDAP relying on the additional `logout:@` data?

Reply:
Also, this operation could probably be done from the controller function `logout401` instead of the view.

Reply:
This is required, in combination with the webserver setup in the accompanying docs, to properly log out Firefox. This is in the view, and as Javascript since the call has to be done *in the users browser* to log out the browser, and not serverside. I will update the commends to make this more clear.

Reply:
Force pushed code with better inline docs and explanation.